### PR TITLE
[DevPortal] Add application creation logics

### PR DESCRIPTION
**Pull request by inthirakumaaran**
This partially fixes the issue https://github.com/wso2/identity-apps/issues/342

### CODE REVIEW

File: apps/admin-portal/src/components/applications/wizard-general-settings.tsx [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+                            <Grid.Row columns={ 1 } className={ "protocolRow" }>
+                                <Grid.Column mobile={ 16 } tablet={ 16 } computer={ 14 } className="">
+                                    <Field
+                                        name="advanceConfiguration"
+                                        label=""
+                                        required={ false }
+                                        requiredErrorMessage="this is needed"
+                                        type="checkbox"
+                                        value={
+                                            applicationData && applicationData.discoverableByEndUsers ? ["discoverableByEndUsers"] : []
```
Comment (jeradrutnam) [inline apps/admin-portal/src/components/applications/wizard-general-settings.tsx]: Fix character length

File: apps/admin-portal/src/components/applications/wizard-protocol-oauth-settings.tsx [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+                                        required={ true }
+                                        requiredErrorMessage="this is needed"
+                                        placeholder="Enter the CallbackURL"
+                                        type="text"
+                                        validation={ (value: string, validation: Validation) => {
+                                            const urlList = value.split(",");
+                                            urlList.map((singleUrl) => {
+                                                if (!FormValidation.url(singleUrl)) {
+                                                    validation.isValid = false;
+                                                    validation.errorMessages.push("Please add valid URLs with comma separation.");
```
Comment (jeradrutnam) [inline apps/admin-portal/src/components/applications/wizard-protocol-oauth-settings.tsx]: Fix character length

File: apps/admin-portal/src/pages/application-template.tsx [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+import { PageLayout } from "../layouts";
+
+/**
+ * Choose the application template from this page.
+ *
+ * @return {JSX.Element}
+ */
+export const ApplicationTemplateSelectPage = (): JSX.Element => {
+
+    // Template list
```
Comment (jeradrutnam) [inline apps/admin-portal/src/pages/application-template.tsx]: Need to localize this. Add a `TODO:` if you planing to do later

Comment (inthirakumaaran) [reply apps/admin-portal/src/pages/application-template.tsx]: Templates will be retrieved from the backend in the future.

File: apps/admin-portal/src/api/application.ts [hunk 4]
```diff
@@ -355,3 +355,31 @@ export const updateAdvanceConfigurations = (id: string, advancedConfigs: object)
...
+            "Content-Type": "application/json"
+        },
+        method: HttpMethods.POST,
+        url: ServiceResourcesEndpoint.applications
+    };
+
+    return httpClient.request(requestConfig)
+        .then((response) => {
+            if (response.status !== 200) {
+                return Promise.reject(new Error("Failed create the application."));
```
Comment (DimalChandrasiri) [inline apps/admin-portal/src/api/application.ts]: IMO the common errors should be in a constance file. Could we add a TODO on this task?

File: apps/admin-portal/src/components/applications/application-templates.tsx [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+    const initiateWizard = (start: boolean) => {
+        if (start) {
+            setWebAppOption(!showWebAppOption);
+        } else {
+            setWizard(true);
+        }
+    };
+
+    const [showWebAppOption, setWebAppOption] = useState(false);
+
```
Comment (DimalChandrasiri) [inline apps/admin-portal/src/components/applications/application-templates.tsx]: Remove the additional new line.

File: apps/admin-portal/src/components/applications/application-templates.tsx [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+    const [showWizard, setWizard] = useState(false);
+
+    const closeWizard = () => {
+        setWizard(false);
+    };
```
Comment (DimalChandrasiri) [inline apps/admin-portal/src/components/applications/application-templates.tsx]: Add a new line after the code block

**Pull request merged**